renderdoc/driver/gl/wrappers/gl_query_funcs.cpp

```diff
@@ -262,8 +262,12 @@ bool WrappedOpenGL::Serialise_glBeginQuery(GLenum target, GLuint qid)
 
   if(m_State < WRITING)
   {
-    m_Real.glBeginQuery(Target, GetResourceManager()->GetLiveResource(id).name);
-    m_ActiveQueries[QueryIdx(Target)][0] = true;
+    // Queries in the log interfere with the queries from FetchCounters.
+    if(!m_FetchCounters)
+    {
+      m_Real.glBeginQuery(Target, GetResourceManager()->GetLiveResource(id).name);
+      m_ActiveQueries[QueryIdx(Target)][0] = true;
```

Comment:
I'm concerned this might interfere with conditional rendering, but I don't see a better alternative honestly.

The only thing I'd say is that because of this, the `GLReplay::FetchCounters` function should do a full replay back to the current event. The assumption with the timers is usually that the internal replay of the captured frame will be identical to the unmodified thing, with extra instrumentation. With this turning off of queries it could be that the rendered results would change and that would be confusing for people.